cmd/ct_monitor/main.go

```diff
@@ -0,0 +1,178 @@
+		}
+	}
+
+	if *configYamlInput != "" {
+		if err := yaml.Unmarshal([]byte(*configYamlInput), &config); err != nil {
+			log.Fatalf("error parsing identities: %v", err)
+		}
+	}
+
+	var fulcioClient *ctclient.LogClient
+	fulcioClient, err := ctclient.New(*serverURL, http.DefaultClient, jsonclient.Options{})
+	if err != nil {
+		log.Fatalf("getting Fulcio client: %v", err)
+	}
+
+	allOIDMatchers, err := config.MonitoredValues.OIDMatchers.RenderOIDMatchers()
+	if err != nil {
+		fmt.Printf("error parsing OID matchers: %v", err)
+	}
+
+	monitoredValues := identity.MonitoredValues{
+		CertificateIdentities: config.MonitoredValues.CertificateIdentities,
+		Subjects:              config.MonitoredValues.Subjects,
+		Fingerprints:          config.MonitoredValues.Fingerprints,
+		OIDMatchers:           allOIDMatchers,
+	}
+
+	for _, certID := range monitoredValues.CertificateIdentities {
+		if len(certID.Issuers) == 0 {
+			fmt.Printf("Monitoring certificate subject %s\n", certID.CertSubject)
+		} else {
+			fmt.Printf("Monitoring certificate subject %s for issuer(s) %s\n", certID.CertSubject, strings.Join(certID.Issuers, ","))
+		}
+	}
+	for _, fp := range monitoredValues.Fingerprints {
+		fmt.Printf("Monitoring fingerprint %s\n", fp)
+	}
+	for _, sub := range monitoredValues.Subjects {
+		fmt.Printf("Monitoring subject %s\n", sub)
+	}
+
+	ticker := time.NewTicker(*interval)
+	defer ticker.Stop()
+
+	// To get an immediate first tick
+	for ; ; <-ticker.C {
+		inputEndIndex := config.EndIndex
+
+		var currentSTH *ctgo.SignedTreeHead
+		if config.StartIndex == nil || config.EndIndex == nil {
+			currentSTH, err = fulcioClient.GetSTH(context.Background())
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "error getting signed tree head: %v", err)
+				return
+			}
+		}
+
+		if config.StartIndex == nil {
+			if config.LogInfoFile != "" {
+				var prevSTH *ctgo.SignedTreeHead
+				prevSTH, err = file.ReadLatestCTSignedTreeHead(config.LogInfoFile)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "reading checkpoint log: %v", err)
+					return
+				}
+
+				checkpointStartIndex := int(prevSTH.TreeSize) //nolint: gosec // G115, log will never be large enough to overflow
+				config.StartIndex = &checkpointStartIndex
+			} else {
+				defaultStartIndex := 0
+				config.StartIndex = &defaultStartIndex
+			}
+		}
+
+		if config.EndIndex == nil {
+			checkpointEndIndex := int(currentSTH.TreeSize) //nolint: gosec // G115
+			config.EndIndex = &checkpointEndIndex
+		}
+
+		if *config.StartIndex >= *config.EndIndex {
+			fmt.Fprintf(os.Stderr, "start index %d must be strictly less than end index %d", *config.StartIndex, *config.EndIndex)
+		}
+
+		_, err = ct.IdentitySearch(fulcioClient, *config.StartIndex, *config.EndIndex, monitoredValues)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to successfully complete identity search: %v", err)
+			return
+		}
+
+		err = ct.RunConsistencyCheck(fulcioClient, config.LogInfoFile)
```

Comment:
Can we flip the order of these, to first check consistency before the identity check?

Reply:
also adding in the current STH as a return value to RunConsistencyCheck to reduce number of calls to GetSTH